src/actions/toggleNote.ts

```diff
@@ -14,23 +20,84 @@ import reducerFlow from '../util/reducerFlow'
 const toggleNote = (state: State): State => {
   const thoughtId = head(state.cursor!)
   const hasNote = findDescendant(state, thoughtId, '=note')
-  const noteChildren = hasNote ? getChildren(state, hasNote) : []
-  const offset = noteChildren.length ? noteChildren[0].value.length : 0
+
+  // Check if there's a path reference (=note/=path)
+  const hasPathReference = hasValidPathReference(state, state.cursor!)
 
   return reducerFlow([
-    // create an empty note if it doesn't exist
+    // Handle path reference: create target thought if needed
+    hasPathReference
+      ? state => {
+          const pathValue = getPathReferenceValue(state, state.cursor!)
+          const targetId = getPathReferenceTargetId(state, state.cursor!)
+
+          // If target doesn't exist but we have a path value, create the target thought
+          if (!targetId && pathValue) {
+            const pathRank = getNextRank(state, thoughtId)
+            return createThought(state, {
+              path: [thoughtId],
+              value: pathValue,
+              rank: pathRank,
+            })
+          }
+          return state
+        }
+      : null,
+
+    // Create an empty child for target if needed
+    hasPathReference
```

Comment:
Which approach is better?

```js
if (A) {
  f()
}
if (A) {
  g()
}
```

or:

```js
if (A) {
  f()
  g()
}
```

You're doing the same thing with `hasPathReference`.